### PR TITLE
reset loop_counter incase a new video is playing

### LINKED ISSUE #78: When we change video, tags change but index doesn't reset
When we call get_config we should also check if the input uri has changed, if it has we should reset the index = 1

### PR DESCRIPTION
**Pull request by Gal-Tch**
a partial fix for issue 78:
https://github.com/ophirhan/cnsr-vlc-viewer-addon/issues/78

### CODE REVIEW

File: intf/cnsr_intf.lua [hunk 1]
```diff
@@ -145,8 +146,21 @@ it runs all the time in the background of VLC, finds the relevant tag and execut
 function looper()
 	local next_loop_time = vlc.misc.mdate()
 	local loop_counter = 0
+	local current_uri = nil
+	
 	while true do
 		if vlc.volume.get() == -256 then break end  -- inspired by syncplay.lua; kills vlc.exe process in Task Manager
+		
+		current_item = vlc.playlist.get(vlc.playlist.current()) -- if a new video is playing, then we want to get the new config
```
Comment (ophirhan) [inline intf/cnsr_intf.lua:154]: Should we maybe check once every get loop interval?
meaning move lines 154-162 all of this code into the if loop_counter  == 0

File: intf/cnsr_intf.lua [hunk 2]
```diff
@@ -145,8 +146,21 @@ it runs all the time in the background of VLC, finds the relevant tag and execut
...
+		if current_item ~= nil and current_uri ~= current_item.path then
+			current_uri = current_item.path
+			tag_by_end_time_index = 1
+			tag_index = 1
+			loop_counter = 0
+			log("new video")
+			sleep(SLEEP_WHILE_PARSING) -- wait for cnsr_ext to finish parsing the new cnsr file
```
Comment (ophirhan) [inline intf/cnsr_intf.lua]: Great idea!
but maybe use vlc.misc.mwait(vlc.misc.mdate() + <microseconds>) which doesn't hog the CPU :-P

**Review by ophirhan — CHANGES_REQUESTED** [review]

**Comment by ishayhil** [comment]
@ophirhan @Gal-Tch loop_counter is removed with #95 PR

**Review by ophirhan — CHANGES_REQUESTED** [review]
Maybe this needs to be re-implemented by adding tag_index = 1
when ever memory.get_written() returns true?

**Review by ophirhan — APPROVED** [review]

**Pull request merged**